internal/common/basic.go

```diff
@@ -204,14 +205,23 @@ func CheckWorkerNodesReady(ctx context.Context, wcClient *client.Client, values
 	minNodes := 0
 	maxNodes := 0
 	for _, pool := range values.NodePools {
+		// MachineDeployment node pool
 		if pool.Replicas > 0 {
 			minNodes += pool.Replicas
 			maxNodes += pool.Replicas
 			continue
 		}
 
-		minNodes += pool.MinSize
-		maxNodes += pool.MaxSize
+		// MachinePool node pool
+		if pool.MinSize > 0 && pool.MaxSize > 0 {
+			minNodes += pool.MinSize
+			maxNodes += pool.MaxSize
+		}
+
+		// Karpenter node pool, we don't care about the number
+		// We set the min to 2 as we have some affinity rules that would require at least that
+		minNodes += 2
+		maxNodes += 99
```

Comment:
Errr... shouldn't this be inside a conditional? Otherwise it's going to increase MachinePools also

Reply:
🤦 Right, let me fix it.